astropy/modeling/core.py

```diff
@@ -2742,18 +2749,40 @@ def _check_param_broadcast(self, max_ndim):
 
         # Now check mutual broadcastability of all shapes
         try:
-            check_broadcast(*all_shapes)
-        except IncompatibleShapeError as exc:
-            shape_a, shape_a_idx, shape_b, shape_b_idx = exc.args
-            param_a = self.param_names[shape_a_idx]
-            param_b = self.param_names[shape_b_idx]
-
-            raise InputParameterError(
-                f"Parameter {param_a!r} of shape {shape_a!r} cannot be broadcast with "
-                f"parameter {param_b!r} of shape {shape_b!r}.  All parameter arrays "
+            np.broadcast_shapes(*all_shapes)
+        except ValueError as exc:
+            base_message = (
+                "All parameter arrays "
                 "must have shapes that are mutually compatible according "
                 "to the broadcasting rules."
             )
+            broadcast_shapes_error_re = re.compile(
+                r"shape mismatch: objects cannot be broadcast to a single shape\.  "
+                r"Mismatch is between "
+                r"arg (?P<argno_a>\d+) with shape (?P<shape_a>\((\d+(, ?)?)+\)) and "
+                r"arg (?P<argno_b>\d+) with shape (?P<shape_b>\((\d+(, ?)?)+\))\."
+            )
+            if (match := broadcast_shapes_error_re.fullmatch(str(exc))) is None:
+                warnings.warn(
```

Comment:
Since parsing the error message emitted by numpy is intrinsically fragile (but probably [our least-bad option](https://github.com/astropy/astropy/issues/15211#issuecomment-2076613999)), I spent extra effort on making this logic as robust as possible: failing to parse an error message should be recoverable.